platform/linux-generic/odp_pool.c

```diff
@@ -1404,6 +1404,55 @@ void odp_pool_print(odp_pool_t pool_hdl)
 	ODP_PRINT("\n");
 }
 
+void odp_pool_print_all(void)
+{
+	uint64_t available;
+	uint32_t i, index, tot, cache_size;
+	uint32_t buf_len = 0;
+	uint8_t type, ext;
+	const int col_width = 24;
+	const char *name;
+	char type_c;
+
+	ODP_PRINT("\nList of all pools\n");
+	ODP_PRINT("-----------------\n");
+	ODP_PRINT(" idx %-*s type   free    tot  cache  buf_len  ext\n", col_width, "name");
+
+	for (i = 0; i < ODP_CONFIG_POOLS; i++) {
+		pool_t *pool = pool_entry(i);
+
+		LOCK(&pool->lock);
+
+		if (!pool->reserved) {
+			UNLOCK(&pool->lock);
+			continue;
+		}
+
+		available  = ring_ptr_len(&pool->ring->hdr);
+		cache_size = pool->cache_size;
+		ext        = pool->pool_ext;
+		index      = pool->pool_idx;
+		name       = pool->name;
+		tot        = pool->num;
+		type       = pool->type;
+
+		if (type == ODP_POOL_BUFFER || type == ODP_POOL_PACKET)
+			buf_len = pool->seg_len;
```

Comment:
For simpler code between lock() - > unlock(), seg_len could be read always and then set buf_len (based on type) after unlock().

Reply:
Fixed in V2.